pkg/scheduler/eventhandlers_test.go

```diff
@@ -334,3 +341,78 @@ func TestPreCheckForNode(t *testing.T) {
+		nodeFunc                 func() *v1.Node
+		existingPods             []*v1.Pod
+		pod                      *v1.Pod
+		flags                    []bool
+		wantNonResource          [][]noderesources.NonResource
+		wantInsufficientResource [][]noderesources.InsufficientResource
+	}{
+		{
+			name: "check nodeAffinity and nodeports, nodeAffinity need fail quickly if includeAllFailures is false",
+			nodeFunc: func() *v1.Node {
+				return st.MakeNode().Name("fake-node").Label("foo", "bar").Obj()
+			},
+			pod: st.MakePod().Name("pod2").HostPort(80).NodeSelector(map[string]string{"foo": "bar1"}).Obj(),
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("pod1").HostPort(80).Obj(),
+			},
+			flags:                    []bool{true, false},
+			wantNonResource:          [][]noderesources.NonResource{{nodeaffinityError, nodeportsError}, {nodeaffinityError}},
+			wantInsufficientResource: [][]noderesources.InsufficientResource{{}, {}},
+		},
+		{
+			name: "check PodOverhead and nodeAffinity, PodOverhead need fail quickly if includeAllFailures is false",
+			nodeFunc: func() *v1.Node {
+				return st.MakeNode().Name("fake-node").Label("foo", "bar").Capacity(cpu).Obj()
+			},
+			pod: st.MakePod().Name("pod2").Container("c").Overhead(v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")}).Req(map[v1.ResourceName]string{v1.ResourceCPU: "1"}).NodeSelector(map[string]string{"foo": "bar1"}).Obj(),
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Req(map[v1.ResourceName]string{v1.ResourceCPU: "7"}).Node("fake-node").Obj(),
+			},
+			flags:                    []bool{true, false},
+			wantNonResource:          [][]noderesources.NonResource{{nodeaffinityError}, nil},
+			wantInsufficientResource: [][]noderesources.InsufficientResource{{podOverheadError}, {podOverheadError}},
+		},
+		{
+			name: "check nodename and nodeports, nodename need fail quickly if includeAllFailures is false",
+			nodeFunc: func() *v1.Node {
+				return st.MakeNode().Name("fake-node").Obj()
+			},
+			pod: st.MakePod().Name("pod2").HostPort(80).Node("fake-node1").Obj(),
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("pod1").HostPort(80).Node("fake-node").Obj(),
+			},
+			flags:                    []bool{true, false},
+			wantNonResource:          [][]noderesources.NonResource{{nodenameError, nodeportsError}, {nodenameError}},
+			wantInsufficientResource: [][]noderesources.InsufficientResource{{}, {}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodOverhead, true)()
+			nodeInfo := framework.NewNodeInfo(tt.existingPods...)
+			nodeInfo.SetNode(tt.nodeFunc())
+
+			for i := range tt.flags {
+				nonResources, insufficientResources := GeneralFilter(tt.pod, nodeInfo, tt.flags[i])
+
+				if diff := cmp.Diff(tt.wantNonResource[i], nonResources); diff != "" {
+					t.Errorf("Unexpected diff in nonResources (-want, +got):\n%s", diff)
```

Comment:
This reads better IMO:

```suggestion
					t.Errorf("Unexpected nonResources (-want, +got):\n%s", diff)
```